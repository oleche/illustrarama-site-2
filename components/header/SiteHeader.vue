<template>
  <b-navbar toggleable="md" class="bg-faded navbar-shadow">
    <b-navbar-toggle target="nav_collapse"></b-navbar-toggle>
    <b-navbar-brand class="d-inline d-sm-none d-xs-none title-brand-sm text-center" href="/">Illustrarama<span class="sub-title-brand"><br/>Source of illustration & design news</span></b-navbar-brand>
    <b-collapse is-nav id="nav_collapse">
      <b-navbar-nav class="mr-auto">
        <b-nav-item href="/">Inicio</b-nav-item>
        <b-nav-item href="/about">Acerca de</b-nav-item>
        <b-button class="btn btn-outline-danger btn-susbcribe" v-b-modal.modal-1 ref="btnShow">Â¡Suscribete!</b-button>
      </b-navbar-nav>
    </b-collapse>
    <b-navbar-brand class="d-none d-lg-block d-xl-block title-brand text-center" href="/">Illustrarama<span class="sub-title-brand"><br/>Source of illustration & design news</span></b-navbar-brand>
    <b-collapse is-nav id="nav_collapse" class="searchfix">
      <b-navbar-nav class="ml-auto">
        <br class="d-block d-sm-none d-xs-none"/>
        <b-nav-form action="/search">
          <b-form-input size="sm" :name="'q'" class="mr-sm-2" type="text" placeholder="Buscar"/>
          <b-button size="sm" class="btn btn-outline-danger my-2 my-sm-0" type="submit">Buscar</b-button>
        </b-nav-form>
      </b-navbar-nav>
    </b-collapse>
  </b-navbar>
</template>
<style scoped>
* {
  font-family: Raleway;
}

.searchfix{
  flex-basis: unset!important;
}

.title-brand{
  font-family: "Poiret One", cursive, sans-serif!important;
  color:#555!important;
  font-weight: bold;
  font-size: 50px;
  line-height: 18px;
  padding-top: 15px;
}

.title-brand-sm{
  font-family: "Poiret One", cursive, sans-serif!important;
  color:#555!important;
  font-weight: bold;
  font-size: 45px;
  line-height: 18px;
  padding-top: 15px;
  margin-right: 0px!important;
}


.sub-title-brand{
  font-family: "Poiret One", cursive, sans-serif!important;
  color:#555!important;
  font-weight: bold;
  font-size: 15px;
  display:block;
  /* margin-top: 30px; */
}

.navbar-shadow{
  box-shadow: -1px -1px 25px #ccc;
  min-height: 95px;
  background-color: #fff;
}

.btn-secondary{
  background-color: #FFF!important;
}
.btn-secondary:hover, .btn-secondary:focus{
  color:#5a6268;
}

.btn-susbcribe{
  /* background: radial-gradient(circle, rgba(220,53,69,1) 52%, rgba(0,0,0,1) 100%); */
  background: white;
}
</style>
<script>
export default {
  name: 'i-header',
}
</script>
