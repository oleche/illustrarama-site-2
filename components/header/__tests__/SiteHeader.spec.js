import { createLocalVue, mount } from '@vue/test-utils'
import SiteHeader from "../SiteHeader"
import BootstrapVue from 'bootstrap-vue'
// import Logo from '@/components/Logo.vue'

const factory = () => {
  const localVue = createLocalVue();
  localVue.use(BootstrapVue);
  return mount(SiteHeader, {
    localVue
  });
};

describe('SiteHeader', () => {
  test('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
})
