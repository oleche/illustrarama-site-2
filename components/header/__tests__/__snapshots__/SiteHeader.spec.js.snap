// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader renders properly 1`] = `
<nav class="navbar bg-faded navbar-shadow navbar-light navbar-expand-md"><button type="button" aria-label="Toggle navigation" aria-controls="nav_collapse" aria-expanded="false" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button> <a target="_self" href="/" class="navbar-brand d-inline d-sm-none d-xs-none title-brand-sm text-center">Illustrarama<span class="sub-title-brand"><br>Source of illustration &amp; design news</span></a>
  <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing">
    <div id="nav_collapse" class="navbar-collapse collapse" style="display: none;">
      <ul class="navbar-nav mr-auto">
        <li class="nav-item"><a target="_self" href="/" class="nav-link">Inicio</a></li>
        <li class="nav-item"><a target="_self" href="/about" class="nav-link">Acerca de</a></li> <button type="button" class="btn btn btn-outline-danger btn-susbcribe btn-secondary">Â¡Suscribete!</button>
      </ul> <a target="_self" href="/" class="navbar-brand d-none d-lg-block d-xl-block title-brand text-center">Illustrarama<span class="sub-title-brand"><br>Source of illustration &amp; design news</span></a>
      <div class="ml-auto right"><br class="d-block d-sm-none d-xs-none">
        <form action="/search" class="navbar-nav"><input name="q" type="text" placeholder="Buscar" class="mr-sm-2 form-control form-control-sm"> <button type="submit" class="btn btn btn-outline-danger my-2 my-sm-0 btn-secondary btn-sm">Buscar</button></form>
      </div>
    </div>
  </transition-stub>
</nav>
`;
